### PR TITLE
Optimize the C++ cold start by decreasing the size of the package

### LINKED ISSUE #711: How come C++ cold-starts are so slow?
I understand rust and go enjoying fast cold starts like:
image

But what could be the reason behind C++ being this slow on the same runtime?
image

Does it have to do with Rust and Go binaries being built with the standard tools? [cargo](https://github.com/maxday/lambda-perf/raw/main/s3-uploader/runtimes/rust_on_provided_al2/Dockerfile) and [go](https://github.com/maxday/lambda-perf/raw/main/s3-uploader/runtimes/go_on_provided_al2/Dockerfile)?  
While the C++ uses an external build tool like [aws-lambda-cpp](https://github.com/awslabs/aws-lambda-cpp)? 

Or is Lambda just less optimized for C++ given it's not used as much?

### PR DESCRIPTION
**Pull request by marcomagdy**
Technically we should be using the `lambda/provided:al2` container image to do this safely.
But since the Lambda function does not do anything except return "OK", There's a good change this might just work.

Fixes #711

### CODE REVIEW

**Comment by marcomagdy** [comment]
After a bit of experimenting with this, it seems like the bulk of the zip file (which is around 5.8MB) is actually libcurl's dynamic libraries.
Excluding the libc runtime brought that size down to 4.7MB, which is an improvement but not large enough.

I got the zip archive to be as small as 1.4MB by building libcurl as a static library and linking the C++ Lambda runtime to it. But I feel that's an advanced usage that most users won't opt for, and therefore it would be misleading representation if we do that here.

So in summary, assuming no ABI breakage between Redhat and AL2, excluding libc is a an easy and small win here.

**Comment by maxday** [comment]
Thanks a lot 🥳 , running a couple of tests tomorrow and it should be live on Monday!
